src/link/Wasm.zig

```diff
@@ -345,11 +345,10 @@ fn linkWithLLD(self: *Wasm, comp: *Compilation) !void {
     // Create an LLD command line and invoke it.
     var argv = std.ArrayList([]const u8).init(self.base.allocator);
     defer argv.deinit();
-    // The first argument is ignored as LLD is called as a library, set it
-    // anyway to the correct LLD driver name for this target so that it's
-    // correctly printed when `verbose_link` is true. This is needed for some
-    // tools such as CMake when Zig is used as C compiler.
-    try argv.append("ld-wasm");
+    // We will invoke ourselves as a child process to gain access to LLD.
+    // This is necessary because LLD does not behave properly as a library -
+    // it calls exit() and does not reset all global data between invocations.
+    try argv.appendSlice(&[_][]const u8{ comp.self_exe_path.?, "wasm-ld" });
```

Comment:
did you mean to swap "ld-wasm" for "wasm-ld"?

Reply:
The symlink that the LLD project installs is `wasm-ld`:

```
andy@ark ~> ls ~/local/llvm11-release/bin/*ld*
/home/andy/local/llvm11-release/bin/ld64.lld
/home/andy/local/llvm11-release/bin/ld.lld
/home/andy/local/llvm11-release/bin/lld
/home/andy/local/llvm11-release/bin/lld-link
/home/andy/local/llvm11-release/bin/llvm-rtdyld
/home/andy/local/llvm11-release/bin/scan-build
/home/andy/local/llvm11-release/bin/wasm-ld
```